### PR TITLE
2012 bug related to species input in goattaxonsearch module

### LINKED ISSUE #2012: bug related to species input in goat/taxonsearch module
### Have you checked the docs?

- [X] [nf-core website: troubleshooting](https://nf-co.re/usage/troubleshooting)
- [X] [nf-core modules documentation](https://nf-co.re/docs/contributing/modules)

### Description of the bug

When using full species names as input only the genus is taken as input, for instance for the European badger, `Meles meles`, this is the output error: 

```
     Command executed:                                                                                                                    
                                                                                                                                              
           goat-cli taxon search \                                                                                                            
               -l -b \                                                                                                                        
               -t Meles meles > test_species.tsv                                                                                              
                                                                                                                                              
           cat <<-END_VERSIONS > versions.yml                                                                                                 
           "test_goat_taxonsearch_genus_name:GOAT_TAXONSEARCH":                                                                               
               goat: $(goat-cli --version | cut -d' ' -f2)                                                                                    
           END_VERSIONS                                                                                                                       
                                                                                                                                              
         Command exit status:                                                                                                                 
           2                                                                                                                                  
                                                                                                                                              
         Command output:                                                                                                                      
           (empty)                                                                                                                            
                                                                                                                                              
         Command error:                                                                                                                       
           error: Found argument 'meles' which wasn't expected, or isn't valid in this context                                                
                                                                                                                                              
           USAGE:                                                                                                                             
               goat-cli taxon search [OPTIONS]                                                                                                
                                                                                                                                              
           For more information try --help           
```
To reproduce the error, only change the taxon `'Canis'` for `'Meles meles'` (line 14) in `tests/modules/goat/taxonsearch/` and then run the following command: 

```
nf-core modules create-test-yml
```


### Command used and terminal output

```console
nf-core modules create-test-yml 

Command executed:                                                                                                                    
                                                                                                                                              
           goat-cli taxon search \                                                                                                            
               -l -b \                                                                                                                        
               -t Meles meles > test_species.tsv                                                                                              
                                                                                                                                              
           cat <<-END_VERSIONS > versions.yml                                                                                                 
           "test_goat_taxonsearch_genus_name:GOAT_TAXONSEARCH":                                                                               
               goat: $(goat-cli --version | cut -d' ' -f2)                                                                                    
           END_VERSIONS                                                                                                                       
                                                                                                                                              
         Command exit status:                                                                                                                 
           2                                                                                                                                  
                                                                                                                                              
         Command output:                                                                                                                      
           (empty)                                                                                                                            
                                                                                                                                              
         Command error:                                                                                                                       
           error: Found argument 'meles' which wasn't expected, or isn't valid in this context                                                
                                                                                                                                              
           USAGE:                                                                                                                             
               goat-cli taxon search [OPTIONS]                                                                                                
                                                                                                                                              
           For more information try --help
```


### Relevant files

[nextflow.log](https://github.com/nf-core/modules/files/9493009/nextflow.log)


### System information

Ubuntu 20.04.5 LTS (Desktop)
Nextflow version 22.04.4 build 5706
nf-core, version 2.5.1

### PR DESCRIPTION
**Pull request by alxndrdiaz**
## PR checklist

Closes #2012 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Comment by alxndrdiaz** [comment]
This PR is part the Google Summer of Code 2022 project: [Conversion of the BlobToolKit pipeline to Nextflow](https://summerofcode.withgoogle.com/myprojects/details/v89lHomV), GitHub repository: https://github.com/sanger-tol/blobtoolkit

**Comment by muffato** [comment]
@alxndrdiaz : can you run prettier on `tests/modules/goat/taxonsearch/test.yml` ?

**Comment by alxndrdiaz** [comment]
> @alxndrdiaz : can you run prettier on `tests/modules/goat/taxonsearch/test.yml` ?

Done.

File: modules/goat/taxonsearch/main.nf [hunk 1]
```diff
@@ -1,6 +1,6 @@
 process GOAT_TAXONSEARCH {
     tag "$meta.id"
-    label 'process_single'
+    label 'process_low'
```
Comment (muffato) [inline modules/goat/taxonsearch/main.nf:3]: This should remain `process_single`. I know `nf-core lint` complains about this, and I've raised a PR: https://github.com/nf-core/tools/pull/1823

Comment (alxndrdiaz) [reply modules/goat/taxonsearch/main.nf]: I changed it back to `process_single`.

**Review by muffato — APPROVED** [review]

**Pull request merged**